### ISSUE TITLE
Cannot use GPU device, error: cudnn64_8.dll not found

### ISSUE THREAD

**Issue by ErinYang96** [Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0#3]
Hello,

I am using the new version of Mask-RCNN to test my code (developed from train_shapes.ipynb), based on TF2.7.0 and keras 2.8.0 as requested. However, when I run my first section, it seems that I cannot use the GPU. I put my main problems and environments as follow.

**My first section code is as follow:**

> import os
> import sys
> import random
> import math
> import re
> import time
> import numpy as np
> import cv2
> import matplotlib
> import matplotlib.pyplot as plt
> import h5py
> from imgaug import augmenters as iaa
> from imgaug import parameters as iap
> from glob import glob
> 
> ROOT_DIR = os.path.abspath('/Mask_RCNN2.0/')
> 
> sys.path.append(ROOT_DIR)  # To find local version of the library
> from mrcnn.config import Config
> from mrcnn import utils
> import mrcnn.model as modellib
> from mrcnn import visualize
> from mrcnn.model import log
> 
> MODEL_DIR = os.path.join(ROOT_DIR, "logs")
> 
> COCO_MODEL_PATH = os.path.join(ROOT_DIR, "mask_rcnn_coco.h5")
> if not os.path.exists(COCO_MODEL_PATH):
>     utils.download_trained_weights(COCO_MODEL_PATH)
> 
> import tensorflow as tf
> config = tf.compat.v1.ConfigProto()
> config.gpu_options.allow_growth=True
> sess = tf.compat.v1.Session(config=config)
> 
**And the error message is as follow:**

> Unable to load extension: pydevd_plugins.extensions.types.pydevd_plugin_pandas_types
> 2022-05-12 18:36:44.911945: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudnn64_8.dll'; dlerror: cudnn64_8.dll not found
> 2022-05-12 18:36:44.913061: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
> To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
> 2022-05-12 18:36:44.913764: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1850] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
> Skipping registering GPU devices...

**My computer setup is as follow (by nvidia-smi.exe):**
GPU: NVIDIA Quadro P620
Driver vesion: 512.15
CUDA Version: 11.6
Note that I don't have any CUDA Toolkit installed from NVIDIA, only the CUDA Driver. 

Do I have to install CUDA Toolkit for using the GPU? Or I shoud install tensorflow-gpu (and the conda version of cudatoolkit and cudnn) ? However, the highest version for tensorflow-gpu is 2.6.0, not 2.7.0. Can I install tensorflow-gpu=2.6.0 to run this code? 

By the way, no such errors emerged when I use tensorflow-gpu=1.5.0. Only now I have to convert to tensorflow 2.x to fit into higher version of CUDA for subsequent training. p.s. My code works fine on Google colab, but not on local machine.

This problem really troubles me for a long time. I would really appreciate it if you can help me figure out where my problem is.

Thanks in advance for your kind support.

Best,
Erin

**Comment by Kamlesh364** [comment]
@ErinYang96 
1. This error is popping up because TensorFlow is not able to access your GPU. 
2. Please setup your GPU as per the instructions provided in https://www.tensorflow.org/install/gpu
3. Tensorflow2 doesn't require tensorflow-gpu to be explicitly installed, so you don't need to install it if you're using any version of Tensorflow2.
4. Without Cuda-toolkit, TensorFlow won't be able to access you GPU, so please follow step 2 properly.

best regards
Kamlesh

**Comment by ErinYang96** [comment]
Hi,

Thanks for answering my questions. Now I have CUDA Toolkit 11.6.2 installed on my computer, but I run into errors like:
`Unable to load extension: pydevd_plugins.extensions.types.pydevd_plugin_pandas_types`

And my GPU seems to be out-of-memory when loading the training dataset, which should be easy (see below).
`RESOURCE_EXHAUSTED: SameWorkerRecvDone unable to allocate output tensor.`
`Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. `

I wonder if it is because my CUDA version is too high? Should I change it to version 11.2 ?

Thanks,
Erin

**Comment by ErinYang96** [comment]
> Hi,
> 
> Thanks for answering my questions. Now I have CUDA Toolkit 11.6.2 installed on my computer, but I run into errors like: `Unable to load extension: pydevd_plugins.extensions.types.pydevd_plugin_pandas_types`
> 
> And my GPU seems to be out-of-memory when loading the training dataset, which should be easy (see below). `RESOURCE_EXHAUSTED: SameWorkerRecvDone unable to allocate output tensor.` `Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. `
> 
> I wonder if it is because my CUDA version is too high? Should I change it to version 11.2 ?
> 
> Thanks, Erin

I have turned my CUDA Toolkit version into 11.2, and got GPU connected it seems. But now I run into another problem, for which I open another issue, see [AttributeError in loading training dataset](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.8.0/issues/4).

Many thanks if you can check this for me.
